evap/static/js/datagrid.js

```diff
@@ -0,0 +1,314 @@
+
+    findCellByColumn(row, column) {
+        return row.children().eq(column);
+    }
+
+    get defaultOrder() {
+        if (this.sortableColumns.length > 0) {
+            return [[this.sortableColumns[0], "asc"]];
+        } else {
+            return [];
+        }
+    }
+}
+
+export class EvaluationGrid extends TableGrid {
+    init({filterButtons, ...options}) {
+        this.filterButtons = filterButtons;
+        super.init(options);
+    }
+
+    bindEvents() {
+        super.bindEvents();
+        this.filterButtons.each((index, element) => {
+            const button = $(element);
+            const count = this.rows.filter(row => {
+                return row.filterValues.get("state").includes(button.data("filter"));
+            }).length;
+            const badgeClass = count === 0 ? "badge-btn-zero" : "badge-btn";
+            button.append(`<span class="badge badge-pill ${badgeClass}">${count}</span>`);
+
+            button.click(() => {
+                if (button.hasClass("active")) {
+                    button.removeClass("active");
+                    this.filter({state: []});
+                } else {
+                    button.addClass("active");
+                    this.filterButtons.not(button).removeClass("active");
+                    this.filter({state: [button.data("filter")]});
+                }
+            });
+        });
+    }
+
+    addFilterRowData(row, data) {
+        const state = row.find(".icon").get()
+            .map(element => $(element).data("filter"));
+        data.filterValues.set("state", state);
+    }
+
+    get defaultOrder() {
+        return [[5, "asc"]];
+    }
+
+    syncFilterStateWithInputs() {
+        super.syncFilterStateWithInputs();
+        if (this.state.filter.state) {
+            this.filterButtons.filter(`[data-filter="${this.state.filter.state[0]}"]`).addClass("active");
+        }
+    }
+}
+
+// Grid based data grid which has its container separated from its header
+export class ResultGrid extends DataGrid {
+    init({sortableHeaders, filterCheckboxes, sortColumnSelect, sortOrderCheckboxes, resetFilter, resetOrder, ...options}) {
+        this.sortableHeaders = new Map();
+        sortableHeaders.each((index, header) => {
+            const column = $(header).data("col");
+            this.sortableHeaders.set(column, $(header));
+        });
+        this.filterCheckboxes = filterCheckboxes;
+        this.sortColumnSelect = sortColumnSelect;
+        this.sortOrderCheckboxes = sortOrderCheckboxes;
+        this.resetFilter = resetFilter;
+        this.resetOrder = resetOrder;
+        super.init(options);
+    }
+
+    bindEvents() {
+        super.bindEvents();
+        for (const [name, {checkboxes}] of Object.entries(this.filterCheckboxes)) {
+            checkboxes.on("change", () => {
+                const values = checkboxes.filter(":checked").get().map(elem => elem.value);
+                this.filter({[name]: values});
```

Comment:
why is `name` in a list here? that doesn't seem to be the case for the other calls to `filter`

Reply:
I changed `filter` to be a map, not a object. This removes some quirky spots in the code, like the `Object.assign` and the `[key]` syntax here (the difference between `{key: value}` and `{[key]: value}` is that the key of the latter will be dynamically determined by a variable)